src/locators/vyperLocator.js

```diff
@@ -0,0 +1,48 @@
+var ControlLocator = require('./controlLocator');
+var VyperParser = require('../vyperParser');
+
+/**
+ * Example of how to register a new locator that
+ * accepts an arbitrary JSON, translates it to an OPA5 declarative JSON object, and uses the internal logic of by.control to find a control
+ * @constructor
+ * @param {Config} config
+ * @param {Object} instanceConfig
+ * @param {Logger} logger
+ */
+function VyperLocator(config, instanceConfig, logger) {
+  this.config = config;
+  this.instanceConfig = instanceConfig;
+  this.logger = logger;
+}
+
+/**
+ * Register custom locator
+ * @param {ProtractorBy} by protractor By object on which to add the new locator
+ */
+VyperLocator.prototype.register = function (by) {
+  this.logger.debug('Registering custom locator');
+  by.ui5All = function(mMatchers) {
```

Comment:
this is the way that is writen currently, it will not change yet but improvement will be considered (unfortunetaly was called like that always before we got it handover to us)